android/app/src/main/java/org/fossasia/openevent/core/notifications/repository/NotificationsRepository.java

```diff
@@ -0,0 +1,37 @@
+package org.fossasia.openevent.core.notifications.repository;
+
+import android.arch.lifecycle.MutableLiveData;
+
+import org.fossasia.openevent.common.api.APIClient;
+import org.fossasia.openevent.common.api.JWTUtils;
+import org.fossasia.openevent.core.auth.AuthUtil;
+import org.fossasia.openevent.data.repository.RealmDataRepository;
+import org.json.JSONException;
+
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+import timber.log.Timber;
+
+public class NotificationsRepository {
+
+    public static Disposable downloadNotifications(MutableLiveData<Boolean> notificationDownloadResponse) {
```

Comment:
Don't make the method static, it creates problems during testing, and return an Observable and not a disposable. Responsibility of subscribing is on the caller

Reply:
Also. the Repository should have no idea about LiveData, only unidirectional flow of data should be present. Thus once you return Observable, you won't require LiveData